pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -21,7 +21,7 @@ import (
 const (
 	defaultTenantLabel   = "__tenant_id__"
 	retainExistingPrefix = "original_"
-	maxConcurrency       = 16
+	maxConcurrency       = 60
```

Comment:
What effect does this have?
Please ensure that changes which impact even when not using the feature are called out in the PR description, maybe in the changelog.

Reply:
how about making this one a flag? we can use this param could to tune the maximum concurrent fanout when executing cross-tenant queries or rules.

Reply:
This change increases the federated tenant query max concurrent queries from 16 to 60. 
I like the idea of making this a flag/config parameter and will proceed with that, leaving the default value at 16 if that sounds ok